codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -335,6 +340,234 @@ private void writeSafeFieldAccessor(
+                .filter(entry -> isRestBinding(entry.getValue().getLocation()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (v1, v2) -> {
+                    throw new CodegenException("found duplicate binding entries for same error shape");
+                }, TreeMap::new));
+
+        // do not generate if no HTTPBinding for Error Binding
+        if (bindingMap.size() == 0) {
+            return;
+        }
+
+        generateShapeDeserializerFunction(writer, model, symbolProvider, errorBindingShape, bindingMap);
+    }
+
+
+    private void generateShapeDeserializerFunction(
+            GoWriter writer,
+            Model model,
+            SymbolProvider symbolProvider,
+            Shape targetShape,
+            Map<String, HttpBinding> bindingMap
+    ) {
+        Symbol targetSymbol = symbolProvider.toSymbol(targetShape);
+        Symbol smithyHttpResponsePointableSymbol = SymbolUtils.createPointableSymbolBuilder(
+                "Response", GoDependency.SMITHY_HTTP_TRANSPORT).build();
+
+        writer.addUseImports(GoDependency.FMT);
+
+        String functionName = ProtocolGenerator.getOperationDeserFunctionName(targetSymbol, getProtocolName());
+        writer.openBlock("func $L(v $P, response $P) error {", "}",
+                functionName, targetSymbol, smithyHttpResponsePointableSymbol,
+                () -> {
+                    writer.openBlock("if v == nil {", "}", () -> {
+                        writer.write("return fmt.Errorf(\"unsupported deserialization for nil %T\", v)");
+                    });
+                    writer.write("");
+
+                    for (Map.Entry<String, HttpBinding> entry : bindingMap.entrySet()) {
+                        HttpBinding binding = entry.getValue();
+                        writeRestDeserializerMember(writer, model, symbolProvider, binding);
+                        writer.write("");
+                    }
+                    writer.write("return nil");
+                });
+    }
+
+
+    private void addErrorShapeBinders(GenerationContext context, OperationShape operation) {
+        for (ShapeId errorBinding: operation.getErrors()) {
+            serializeErrorBindingShapes.add(errorBinding);
+        }
+    }
+
+
+    private String generateHttpBindingsValue(
+            GoWriter writer,
+            Model model,
+            Shape targetShape,
+            HttpBinding binding,
+            String operand
+    ) {
+        String value = "";
+        switch (targetShape.getType()) {
+            case STRING:
+                if (targetShape.hasTrait(EnumTrait.class)) {
+                    value = String.format("types.%s(%s)", targetShape.getId().getName(), operand);
+                    return value;
+                }
+                return operand;
+            case BOOLEAN:
+                writer.addUseImports(GoDependency.STRCONV);
+                return String.format("strConv.ParseBool(%s)", operand);
```

Comment:
nit `strconv` probably should be `strconv`. Minor but we should also probably be using symbols for all of these imports to make sure they are imported, and use correctly. Especially in cases of import aliases.

Reply:
There are a couple other instances of `strConv` in here.

Reply:
updated to `strconv`. Not changing to use formatter or symbol as this is static text and won't change. We can come back to this, if we see more value in making these symbols. 

For aliasing we will generally have same alias throughout that are defined in GoDependency enums. So we must use those aliases directly.